Source/WebCore/html/canvas/EXTDisjointTimerQuery.cpp

```diff
@@ -31,7 +31,7 @@
 #include "EventLoop.h"
 #include "ScriptExecutionContext.h"
 #include "WebGLRenderingContext.h"
-
+#include <JavaScriptCore/TypedArrayInlines.h>
```

Comment:
If new rule is in effect, then it is more consistent to have this include not in here but in new file WebGLAnyInlines.h. So all the .cpp files that include WebGLAny include also WebGLAnyInlines. However, I don't see it being correct, as any transitive includes does not know to do it.

Reply:
And to expand: this change needs to be done to all .cpp files that include WebGLAny.h. If they compile now, it's just matter of luck, either by some other header including the inlines or some other .cpp including them and being in same unified build bundle.

Reply:
> If new rule is in effect, then it is more consistent to have this include not in here but in new file WebGLAnyInlines.h. So all the .cpp files that include WebGLAny include also WebGLAnyInlines.

Yes exactly. This is precisely the point. WebGLAnyInlines.h must be included anywhere that WebGLAny methods are _used_ and not merely _referenced_.

> However, I don't see it being correct, as any transitive includes does not know to do it.

And this is also precisely the point. Compilation units which _use_ WebGLAny rather than merely _refer_ to it cannot depend on some transitive header including the implementation of WebGLAny.

> And to expand: this change needs to be done to all .cpp files that include WebGLAny.h from some include chain. If they compile now, it's just matter of luck, either by some other header including the inlines or some other .cpp including them and being in same unified build bundle.

That is strictly false, as there was no use of that inlined method before 260367@main, and it was only used from a single .cpp file added by that same commit.

Reply:
And to expand on that point, _every type_ referenced in WebGLAny.h is only forward declared (and not fully defined) within that header. So _every_ user of WebGLAny.h _must_ also include the necessary definitions for those particular uses. So no, there is not a compile problem lurking within the Unified Build, and this is not some new requirement imposed upon WebGLAny.h.